lab-dustin/route/gallery.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+const Router = require('express');
+const jsonParser = require('body-parser').json();
+const createError = require('http-errors');
+const debug = require('debug')('app:route/gallery');
+const Gallery = require('../model/gallery');
+const bearerAuth = require('../lib/bearer-auth-middleware');
+
+const router = module.exports = new Router();
+
+router.get('/api/gallery/:id',bearerAuth,(req,res,next) => {
+  debug(`GET /api/gallery/${req.params.id}`);
+
+  Gallery.findById(req.params.id)
+    .then(gallery => {
+      if (!gallery) {
+        debug('Gallery not found');
+        return next(createError(404,'Gallery not found'));
```

Comment:
This should probably `Promise.reject(creatError(...))` instead of calling `next()`.